query-languages/m/m-spec-basic-concepts.md

```diff
@@ -152,9 +152,9 @@ The simplest form of identifier reference is an _exclusive-identifier-reference_
 _exclusive-identifier-reference:<br/>
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;identifier_
 
-It is an error for an _exclusive-identifier-reference_ to refer to a variable that is not part of the environment of the expression that the identifier appears within, or to refer to an identifier that is currently being initialized.
```

Comment:
Why is the "or to refer..." part being removed? References like "Section1!Foo" aren't exclusive-identifier-references. They're section-access-expressions.

Reply:
Good point. 

What about this reference, though?

![image](https://user-images.githubusercontent.com/857202/185200533-d9abd25e-f589-48cd-8a4a-b6716bf39b31.png)

It's an _exclusive-identifier-reference_ and is referring to an identifier that is being initialized. However, the "or refer to an identifier that is currently being initialized" restriction doesn't apply to it since the referenced identifier isn't in a let expression or record.

Reply:
Ah, good point. I had forgotten that query self-references don't require @ (which is probably a bug, but that ship has sailed). This means your new wording is correct.